test/test_releasenotes.py

```diff
@@ -0,0 +1,113 @@
+
+
+logging.basicConfig()
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+
+
+class ReleaseNotesTest(unittest.TestCase):
+    def setUp(self):
+        self.release = ReleaseNotes(FAKE_RELEASE_NOTES)
+
+    def test_parse_current_release_notes(self):
+        actual_release = ReleaseNotes(self.release.DEFAULT_FILE)
+        component = actual_release.component
+        self.assertIsNotNone(component)
+
+    def test_get_component_name(self):
+        name = self.release.component_name
+        self.assertEqual(name, "Stream Engine")
+
+    def test_get_latest_version(self):
+        version_number = self.release.latest_version
+        self.assertEqual(version_number, "1.10.0")
+
+    def test_get_latest_date(self):
+        date = self.release.latest_date
+        self.assertEqual(date, "2019-07-19")
+
+    def test_get_latest_descriptor(self):
+        descriptor = self.release.latest_descriptor
+        self.assertEqual(descriptor, "Development Release")
+
+    def test_get_component(self):
+        component = self.release.component
+        self.assertIsNotNone(component)
+
+    def test_get_dependencies(self):
+        dependencies = self.release.dependencies
+        package_names = [dependency.split("=")[0] for dependency in dependencies]
+        self.assertIsNotNone(dependencies)
+        self.assertIn("mock", package_names)
+
+    def test_component_serialization(self):
+        component = self.release.component
+        serialized_component = json.dumps(component, cls=ComponentDecoder, indent=2, separators=(',', ': '))
+        self.assertIsNotNone(serialized_component)
+
+    def test_known_version(self):
+        component = self.release.component
+        version = component.versions[4]
+        self.assertEqual(version.descriptor, "Release")
+        self.assertEqual(version.number, "1.6.0")
+        self.assertEqual(version.date, "2018-07-20")
+        self.assertIsNone(version.details)
+        self.assertIs(len(version.notes), 5)
+
+    def test_known_note(self):
+        component = self.release.component
+        version = component.versions[4]
+        note = version.notes[0]
+        self.assertEqual(note.title, "Issue #13448 - Prevent getting fill values for wavelength coordinate")
+        self.assertEqual(note.ticket, "13448")
+        self.assertEqual(note.details, "- Correct code error that led to NaNs for wavelength values\n- Add handling "
+                                       "to replace coordinate fill values with whole number sequence")
+
+    def test_version_without_date(self):
+        component = self.release.component
+        version = component.versions[18]
+        self.assertEqual(version.number, "1.3.0")
+        self.assertEqual(version.date, "")
+
+    def test_version_without_notes(self):
+        component = self.release.component
+        version = component.versions[18]
+        self.assertEqual(version.number, "1.3.0")
+        self.assertFalse(version.notes)
```

Comment:
notes should be returning a list:
`self.assertListEqual(version.notes, [])`
or
`self.assertEqual(len(version.notes), 0)`

Reply:
The assertFalse successfully checks that I do not have a populated list (an if check on an empty list returns false), but it does not check that the version.notes is a list (vs say a boolean). Opting for the first assert statement you list.